src/main/java/org/gridsuite/explore/server/services/StudyService.java

```diff
@@ -6,138 +6,117 @@
-                .retrieve()
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.add(HEADER_USER_ID, userId);
+        HttpEntity<Map<String, Object>> request = new HttpEntity<>(
+                importParams, headers);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, request, Void.class);
+        } catch (HttpStatusCodeException e) {
+            throw new ExploreException(INSERT_STUDY_FAILED);
+        }
     }
 
-    public Mono<Void> insertStudyWithCaseFile(UUID studyUuid, String userId, Mono<FilePart> caseFile) {
-        return caseFile.flatMap(file -> {
-            MultipartBodyBuilder multipartBodyBuilder = new MultipartBodyBuilder();
-            multipartBodyBuilder.part("caseFile", file);
-
-            String path = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION +
-                            "/studies?studyUuid={studyUuid}")
-                    .buildAndExpand(studyUuid)
-                    .toUriString();
-
-            return webClient.post()
-                    .uri(studyServerBaseUri + path)
-                    .header(HEADER_USER_ID, userId)
-                    .header(HttpHeaders.CONTENT_TYPE, MediaType.MULTIPART_FORM_DATA.toString())
-                    .body(BodyInserters.fromMultipartData(multipartBodyBuilder.build()))
-                    .retrieve()
-                    .bodyToMono(Void.class)
-                    .publishOn(Schedulers.boundedElastic())
-                    .log(ROOT_CATEGORY_REACTOR, Level.FINE);
-        });
+    public void insertStudyWithCaseFile(UUID studyUuid, String userId, MultipartFile caseFile) {
+        MultipartBodyBuilder multipartBodyBuilder = new MultipartBodyBuilder();
+        String path = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION +
+                        "/studies?studyUuid={studyUuid}")
+                .buildAndExpand(studyUuid)
+                .toUriString();
+        try {
+            multipartBodyBuilder.part("caseFile", caseFile.getBytes()).filename(Objects.requireNonNull(caseFile.getOriginalFilename()));
+        } catch (IOException e) {
+            throw new ExploreException(IMPORT_CASE_FAILED);
+        }
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.MULTIPART_FORM_DATA);
+        headers.add(HEADER_USER_ID, userId);
+        HttpEntity<MultiValueMap<String, HttpEntity<?>>> request = new HttpEntity<>(
+                multipartBodyBuilder.build(), headers);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, request, Void.class);
+        } catch (HttpStatusCodeException e) {
+            throw new ExploreException(INSERT_STUDY_FAILED);
```

Comment:
remove this try catch that was not here before to handle the exception with the exception handler ?

Reply:
done